packages/budget-it/src/components/ExpenseItem.vue

```diff
@@ -1,52 +1,53 @@
 <script setup lang="ts">
   import { computed, ref } from 'vue';
-  import type { ExpenseInfo } from '@/stores/expenses';
-  import { useExpensesStore } from '@/stores/expenses';
-  import { useSourcesStore } from '@/stores/sources';
   import { AppInputNumber, AppDropdown } from '@inkbeard/ui-vue';
 
-  const props = defineProps<{
-    expense: ExpenseInfo;
-  }>();
-  const { expenseList } = useExpensesStore();
-  const expenseAmount = ref(props.expense.amount);
-  const expenseSourceId = computed({
-    get: () => props.expense.sourceId,
-    set: (value: number) => {
-      const { id } = props.expense;
+  export interface ExpenseInfo {
+    amount: number // Amount of the expense
+    categoryId: number // Unique identifier for the category the expense belongs to`
+    description?: string // Description of the expense
+    id: number // Unique identifier for the expense
+    name: string // Name of the expense
+    order: number // Order of the expense in the list
+    sourceId: number // Source Id of the expense that is mapped from the sources store
+  }
 
-      if (id) {
-        expenseList[id].sourceId = value;
-      }
-    },
-  });
-  const updateExpenseAmount = () => {
-    const { id } = props.expense;
+  export interface ExpenseList {
+    [key: number]: ExpenseInfo;
+  }
 
-    if (id) {
-      expenseList[id].amount = expenseAmount.value;
-    }
-  };
+  const props = defineProps<{
+    sourceList: Record<string, string>;
+  }>();
+  const expense = defineModel<ExpenseInfo>('expense');
+  const expenseAmount = ref(expense.value!.amount);
+  /**
+   * Get an alphabatize list of sources and their IDs.
+   */
+  const alphabaticSourceList = computed(() => Object.entries(props.sourceList)
+    .map(([id, source]) => ({ source, id: +id }))
+    .sort((a, b) => a.source.toLowerCase().localeCompare(b.source.toLowerCase())));
 
 </script>
 
 <template>
   <form @submit.prevent>
     <AppInputNumber
-      :id="`${expense.categoryId}-${expense.name}`"
+      :id="`${expense!.categoryId}-${expense!.name}`"
```

Comment:
I was getting an error saying these keys didn't exists on `expense`, but the type is saying they're required. I think I'm missing something with the typepscript that I don't understand yet. For now, I'm adding the `!` to tell it that these will exist.